### PR TITLE
[docs] expand TypeSheet.md

### PR DESCRIPTION
**Pull request by anjakefala**
@reagle Please also review. =)

### CODE REVIEW

File: visidata/guides/TypesSheet.md [hunk 1]
```diff
@@ -10,4 +10,27 @@ Columns usually begin as untyped (`anytype`), but can be set.
...
 
-*Note*: `float` uses the decimal separator; `currency` removes non-numeric characters and parses as float; `date` parses dates into ISO8601 format; `len` formats the cell value to the length of the content.
\ No newline at end of file
+## Type formatting
+
+VisiData pre-set defaults for formatting types:
+
+- `currency` removes non-numeric characters and parses as `float`
+- `date` parses dates into ISO8601 format
+- `float` uses the decimal seperator, keeping two significant digits
```
Comment (saulpw) [inline visidata/guides/TypesSheet.md]: mention the command(s) to chance precision?

File: visidata/guides/TypesSheet.md [hunk 2]
```diff
@@ -10,4 +10,27 @@ Columns usually begin as untyped (`anytype`), but can be set.
...
 - {help.commands.type-string}
 
-*Note*: `float` uses the decimal separator; `currency` removes non-numeric characters and parses as float; `date` parses dates into ISO8601 format; `len` formats the cell value to the length of the content.
\ No newline at end of file
+## Type formatting
+
+VisiData pre-set defaults for formatting types:
+
+- `currency` removes non-numeric characters and parses as `float`
+- `date` parses dates into ISO8601 format
```
Comment (saulpw) [inline visidata/guides/TypesSheet.md]: "from any format into date object (shown as ISO8601)"?

File: visidata/guides/TypesSheet.md [hunk 3]
```diff
@@ -10,4 +10,27 @@ Columns usually begin as untyped (`anytype`), but can be set.
...
+- `float` uses the decimal seperator, keeping two significant digits
+- `len` formats the cell value to the length of the content
```
Comment (saulpw) [inline visidata/guides/TypesSheet.md]: officially it's `vlen`.

File: visidata/guides/TypesSheet.md [hunk 4]
```diff
@@ -10,4 +10,27 @@ Columns usually begin as untyped (`anytype`), but can be set.
...
+
+- {help.options.disp_currency_fmt}
+- {help.options.color_currency_neg}
+- {help.options.disp_date_fmt}
+- {help.options.disp_float_fmt}
+- {help.options.disp_int_fmt}
+
+## Importance of typing
+
+Certain commands in VisiData behave differently depending on how the column is typed.
```
Comment (saulpw) [inline visidata/guides/TypesSheet.md]: Remember, we don't need to say "in VisiData" inside VisiData.

File: visidata/guides/TypesSheet.md [hunk 5]
```diff
@@ -10,4 +10,27 @@ Columns usually begin as untyped (`anytype`), but can be set.
...
+Grouping by a numeric column in a Frequency table can result in numeric range binning. Grouping by a string results in categorical binning.
+
+Un-typed columns (`anytype`) often default to `string`-like behaviour. This can lead to confusing results with some commands. For example. `sort-col` defaults to traditional lexigraphic ordering, and `exec-col col1 + col2` results in a concatenation of the two columns, instead of addition.
```
Comment (saulpw) [inline visidata/guides/TypesSheet.md]: I think you mean `addcol-expr`?  Let's break this into multiple lines, anyway.

**Comment by reagle** [comment]
All this looks fine to me.

File: visidata/guides/TypesSheet.md (hunk 3 above)
Comment (anjakefala) [reply visidata/guides/TypesSheet.md]: Should I adjust the docstrings?

visidata/metasheets.py:ColumnsSheet.addCommand('gz#', 'type-len-selected', 'onlySelectedRows.type=vlen', 'set type of selected source columns to len')

**Pull request merged**